### ISSUE TITLE
[Question] ASR Transformer performance vs. Google Speech-to-Text

### ISSUE THREAD

**Issue by mabergerx** [tensorflow/tensor2tensor#1121]
### Description
We used the ["ASR with Transformer" colab notebook](https://colab.research.google.com/github/tensorflow/tensor2tensor/blob/master/tensor2tensor/notebooks/asr_transformer.ipynb) which let us load the pre-trained checkpoints of the ASR Problems trained on librispeech and Common Voice datasets. We tried out a few sentences and the results were not very good, for example for librispeech:

Target: "Hello world" 
Output: "HALLOW WORLDS"

Target: "To which address can we send the official documents?"
Output: "THE WITCH ANNA IS COMING SCENT OFFICIAL LOGAMENTS"

If we compare this to the performance of the Google Cloud Speech-to-Text API (and the service used in Google Translate from voice, which I assume is the same API), that performance is very, very good. In the [paper](https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/46687.pdf), the architecture used is a encoder-decoder one with attention, just as transformer. However a separate  language model is used. Does that make such a huge difference? Or are the checkpoints in the colab notebook not trained on such amount of data / for so long as the Speech-to-Text API?

In general, would it be possible to achieve a result which is better than the result in the ASR colab notebook with the Transformer architecture (and for different languages?)?  

Thanks!

**Comment by nshmyrev** [comment]
No, state of the art systems use much more algorithmic features. The notebook is just a demo for beginners.

**Comment by cwlinghk** [comment]
I found most of the available models on github are still far from google/microsoft/apple's speech to text  performance. What is missing? Training data or language model??

**Comment by cantwbr** [comment]
There is a plethora of methods to improve the speech2text performance. Have a look at [wer_are_we
](https://github.com/syhw/wer_are_we) to find the best performing methods. 
A summary of useful methods include: 
- Huge training sets (The authors of [this paper](https://arxiv.org/abs/1811.06621) use more than 27k hours of speech)
- Spectrum Augmentation (An implementation for t2t can be found here: https://github.com/Kyubyong/specAugment)
- Extended text encodings like chenones (Described [here](https://arxiv.org/abs/1910.01493)) and/or subword unit encoding (Described [here](https://www.isca-speech.org/archive/Interspeech_2018/abstracts/1616.html) for instance)
- Language models applied in the decoder step

to name just a few methods. Feel free to add more ideas.

**Comment by cogmeta** [comment]
How can we apply language model in the t2t decoder process?
